api/views/enrollment_test.py

```diff
@@ -16,4 +24,24 @@ class EnrollmentTestViewSet(ReadOnlyModelViewSet[EnrollmentTest]):
 
 class EnrollmentTestResultViewSet(CreateModelMixin, GenericViewSet[EnrollmentTestResult]):
     queryset = EnrollmentTestResult.objects.all()
-    serializer_class = EnrollmentTestResultCreateSerializer
+
+    def get_serializer_class(self) -> type[BaseSerializer[EnrollmentTestResult]]:
+        if self.action == "get_level":
+            return EnrollmentTestResultLevelSerializer
+        return EnrollmentTestResultCreateSerializer
+
+    @action(detail=False, methods=["post"])
+    def get_level(self, request: Request) -> Response:
+        """Calculates level of language based on number of correct answers."""
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        answer_ids = cast(list[int], request.POST.getlist("answers"))
+        try:
+            return Response(EnrollmentTestResultLevelSerializer.calculate_level(answer_ids))
+        except NotImplementedError:
+            # Repeating the message rather than putting str(e), otherwise CodeQL complains about
```

Comment:
I think we could remove this comment, it's pretty clear why we do any rewording here

Reply:
Done